dotCMS/src/main/java/com/dotcms/graphql/DotGraphQLContext.java

```diff
@@ -1,23 +1,121 @@
 package com.dotcms.graphql;
 
 import com.liferay.portal.model.User;
+import graphql.kickstart.execution.context.DefaultGraphQLContext;
+import graphql.kickstart.servlet.context.GraphQLServletContext;
+import org.dataloader.DataLoaderRegistry;
 
+import javax.security.auth.Subject;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-import graphql.servlet.GraphQLContext;
-
-public class DotGraphQLContext extends GraphQLContext {
+/**
+ * Extends the {@link DefaultGraphQLContext} to be able to set the dotCMS user and have it
+ * available from the different {@link graphql.schema.DataFetcher}s
+ */
+public class DotGraphQLContext extends DefaultGraphQLContext implements
+        GraphQLServletContext {
 
+    private final HttpServletRequest httpServletRequest;
+    private final HttpServletResponse httpServletResponse;
     private final User user;
 
-    public DotGraphQLContext(final HttpServletRequest httpServletRequest,
-                             final HttpServletResponse httpServletResponse, final User user) {
-        super(httpServletRequest, httpServletResponse);
+    private DotGraphQLContext(DataLoaderRegistry dataLoaderRegistry, Subject subject, HttpServletRequest httpServletRequest,
+            HttpServletResponse httpServletResponse, User user) {
+        super(dataLoaderRegistry, subject);
+        this.httpServletRequest = httpServletRequest;
+        this.httpServletResponse = httpServletResponse;
         this.user = user;
     }
 
+    @Override
+    public HttpServletRequest getHttpServletRequest() {
+        return httpServletRequest;
+    }
+
+    @Override
+    public HttpServletResponse getHttpServletResponse() {
+        return httpServletResponse;
+    }
+
+    @Override
+    public List<Part> getFileParts() {
+        try {
+            return httpServletRequest.getParts().stream()
+                    .filter(part -> part.getContentType() != null)
+                    .collect(Collectors.toList());
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5685344)